sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/TimeWindowUtils.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.feature.management.implementation.timewindow;
+
+import com.azure.spring.cloud.feature.management.implementation.timewindow.recurrence.RecurrenceConstants;
+import org.springframework.util.StringUtils;
+
+import java.time.DayOfWeek;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TimeWindowUtils {
+    public static ZonedDateTime convertStringToDate(String timeStr) {
+        if (!StringUtils.hasText(timeStr)) {
+            return null;
+        }
+        try {
+            return ZonedDateTime.parse(timeStr, DateTimeFormatter.ISO_DATE_TIME);
+        } catch (final DateTimeParseException e) {
+            return ZonedDateTime.parse(timeStr, DateTimeFormatter.RFC_1123_DATE_TIME);
+        }
+    }
+
+    /**
+     * Calculate the number of days that have passed since the start of the week
+     * @param today DayOfWeek enum of today
+     * @param firstDayOfWeek the start day of the week
+     * @return the number of days passed
+     * */
+    public static int daysPassedWeekStart(DayOfWeek today, String firstDayOfWeek) {
```

Comment:
Method names should typically start with a verb. This name seems like a property.

Reply:
Updated to `getPassedWeekDays`